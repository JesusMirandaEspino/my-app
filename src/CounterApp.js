import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value  }  ) => {


    const [ counter, SetCounter ] = useState( value );

    const handleAdd = () => {
        SetCounter( counter + 1 );
    }

    const handleSub = () => {
        SetCounter( counter - 1 );
    }

    const handleReset = () => {
        SetCounter( value );
    }

   // const saludo = 'Hola mundo desde react';

    return (
        <section className="fila">  
            <div className="contenedor1">

                <div className="col-full-12">
                    <h1> CounterApp </h1>
                    <h2>{ value }</h2>
                </div>

                <div className="col-full-12">
                    <h2>{ counter }</h2>
                </div>

                <div className="col-full-12">
                    <button onClick={  handleAdd  } > +1 </button>
                </div>

                <div className="col-full-12">
                    <button onClick={  handleReset  } > Reset </button>
                </div>

                <div className="col-full-12">
                    <button onClick={  handleSub  } > -1 </button>
                </div>
    
            </div>
        </section>
    );
    
}

CounterApp.propTypes = { 
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 10
}

export default CounterApp;
