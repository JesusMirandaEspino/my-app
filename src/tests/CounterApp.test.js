import React from 'react';
import '@testing-library/jest-dom';


import { shallow } from 'enzyme'
import  CounterApp  from '../CounterApp';




describe( ' CounterApp', () => { 


    test( 'Debe de mostar < CounterApp/> correctamente', () => {

        const wrapper = shallow( <CounterApp />);

        expect( wrapper ).toMatchSnapshot();

    });



    test( 'Debe mostrar el counter valor enviado por props', () => {


        const wrapper = shallow( 
            <CounterApp 
                value={ 100 } 
            />);

        const valueCounter = wrapper.find( 'h3' ).text().trim();

        expect( valueCounter  ).toBe( '100' );


    });


} );
