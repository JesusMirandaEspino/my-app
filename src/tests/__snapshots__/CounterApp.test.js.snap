// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` CounterApp Debe de mostar < CounterApp/> correctamente 1`] = `
<section
  className="fila"
>
  <div
    className="contenedor1"
  >
    <div
      className="col-full-12"
    >
      <h1>
         CounterApp 
      </h1>
      <h2>
        10
      </h2>
    </div>
    <div
      className="col-full-12"
    >
      <h3>
        10
      </h3>
    </div>
    <div
      className="col-full-12"
    >
      <button
        onClick={[Function]}
      >
         +1 
      </button>
    </div>
    <div
      className="col-full-12"
    >
      <button
        onClick={[Function]}
      >
         Reset 
      </button>
    </div>
    <div
      className="col-full-12"
    >
      <button
        onClick={[Function]}
      >
         -1 
      </button>
    </div>
  </div>
</section>
`;
